crates/hc_service_check/src/main.rs

```diff
@@ -0,0 +1,355 @@
+                Cmd::Offer(offer) => {
+                    c2.set_remote_description(offer).await.unwrap();
+                    let mut answer = c2.create_answer(AnswerConfig::default()).await.unwrap();
+                    println!(
+                        "created answer: {:?}",
+                        String::from_utf8_lossy(&answer.to_vec().unwrap())
+                    );
+                    c2.set_local_description(&mut answer).await.unwrap();
+                    t2o_snd.send(Cmd::Answer(answer)).unwrap();
+                    if let Some(ice_buf) = ice_buf.take() {
+                        for ice in ice_buf {
+                            c2.add_ice_candidate(ice).await.unwrap();
+                        }
+                    }
+                }
+                Cmd::Ice(ice) => {
+                    if let Some(ice_buf) = ice_buf.as_mut() {
+                        ice_buf.push(ice);
+                    } else {
+                        c2.add_ice_candidate(ice).await.unwrap();
+                    }
+                }
+                oth => panic!("unexpected: {oth:?}"),
+            }
+        }
+
+        Ok(())
+    })
+    .await
+    .map_err(Error::other)?
+}
+
+fn print_chan_ready_time(start: std::time::Instant) {
+    static CR: std::sync::Once = std::sync::Once::new();
+    CR.call_once(move || {
+        let elapsed = start.elapsed().as_secs_f64();
+        println!("\nchan ready in {elapsed} seconds");
+    });
+}
+
+fn print_rcv_done_time(start: std::time::Instant) {
+    static RD: std::sync::Once = std::sync::Once::new();
+    RD.call_once(move || {
+        let elapsed = start.elapsed().as_secs_f64();
+        println!("\nreceive done in {elapsed} seconds");
+    });
+}
+
+async fn spawn_peer(
+    config: PeerConnectionConfig,
+) -> (
+    PeerConnection,
+    tokio::sync::mpsc::UnboundedReceiver<PeerConnectionEvent>,
+) {
+    let (con, rcv) = PeerConnection::new(config).await.unwrap();
+    (con, rcv)
+}
+
+async fn spawn_chan(
+    data_chan: DataChannel,
+    mut data_recv: tokio::sync::mpsc::UnboundedReceiver<DataChannelEvent>,
+    start: std::time::Instant,
+    chan_ready: Arc<tokio::sync::Barrier>,
+    rcv_done: Arc<tokio::sync::Barrier>,
+) {
+    loop {
+        match data_recv.recv().await {
+            Some(DataChannelEvent::Open) => break,
+            Some(DataChannelEvent::BufferedAmountLow) => (),
+            oth => panic!("{oth:?}"),
+        }
+    }
+
+    println!("chan ready");
+
+    chan_ready.wait().await;
+
+    print_chan_ready_time(start);
+
+    for _ in 0..MSG_CNT {
+        let buf = GoBuf::from_slice(ONE_KB).unwrap();
+        data_chan.send(buf).await.unwrap();
+    }
+
+    let mut cnt = 0;
+
+    loop {
+        match data_recv.recv().await {
+            Some(DataChannelEvent::Open) => (),
+            Some(DataChannelEvent::BufferedAmountLow) => (),
+            Some(DataChannelEvent::Message(mut buf)) => {
+                assert_eq!(1024, buf.len().unwrap());
```

Comment:
this would make the association more clear

```suggestion
                assert_eq!(ONE_KB.len(), buf.len().unwrap());
```